jwt/jwk.py

```diff
@@ -279,25 +276,139 @@ def jwk_from_dict(dct: Mapping[str, Any]) -> AbstractJWKBase:
+                'content must be bytes, it is {}'.format(type(content)))
+        load_function = f'serializer_load_function_{kind}'
+        if kwargs.get(load_function) is None:
+            raise Exception(f'{load_function} is not Optional')
+        # now convert it to a Callable if it's a string
+        if isinstance(kwargs.get(load_function), str):
+            kwargs[load_function] = getattr(
+                serialization_module,
+                kwargs.get(load_function),
+            )
+
+        if kwargs.get('options') is None:
+            kwargs['options'] = {}
+
+        return func(content, **kwargs)
+    return wrapper
+
+
+@jwk_from_bytes_argument_conversion
+def jwk_from_private_bytes(
+    content: bytes,
+    *,
+    serializer_load_function_private:
+        Union[str, Callable[[bytes, Optional[str], object], object]] = None,
+    password: Optional[str] = None,
+    backend: Optional[object] = None,
+    options: Optional[dict] = None,
+) -> Optional[AbstractJWKBase]:
+    """This function is meant to be called from jwk_from_bytes"""
     try:
-        privkey = load_pem_private_key(
-            pem_content, password=None, backend=default_backend())
+        privkey = serializer_load_function_private(
+            content,
+            password=password,
+            backend=backend,
+        )
         if isinstance(privkey, RSAPrivateKey):
-            return RSAJWK(privkey)
-        raise UnsupportedKeyTypeError(
-            'unsupported key type')  # pragma: no cover
+            return RSAJWK(privkey, **options)
+        raise UnsupportedKeyTypeError('unsupported key type')
     except ValueError:
-        pass
-
+        return None
+
+
+@jwk_from_bytes_argument_conversion
+def jwk_from_public_bytes(
+    content: bytes,
+    *,
+    serializer_load_function_public:
+        Union[str, Callable[[bytes, Optional[str], object], object]] = None,
+    backend: Optional[object] = None,
+    options: Optional[dict] = None
+) -> Optional[AbstractJWKBase]:
+    """This function is meant to be called from jwk_from_bytes"""
     try:
-        pubkey = load_pem_public_key(pem_content, backend=default_backend())
+        pubkey = serializer_load_function_public(
+            content,
+            backend=backend,
+        )
         if isinstance(pubkey, RSAPublicKey):
-            return RSAJWK(pubkey)
+            return RSAJWK(pubkey, **options)
         raise UnsupportedKeyTypeError(
             'unsupported key type')  # pragma: no cover
     except ValueError as why:
-        raise UnsupportedKeyTypeError('could not deserialize PEM') from why
+        raise UnsupportedKeyTypeError('could not deserialize') from why
+
+
+def jwk_from_bytes(
+    content: bytes,
+    *,
+    serializer_load_function_private:
+        Union[str, Callable[[bytes, Optional[str], object], object]] = None,
+    serializer_load_function_public:
+        Union[str, Callable[[bytes, Optional[str], object], object]] = None,
```

Comment:
that str is the name of a function from `cryptography.hazmat.primitives.serialization` module
otherwise you can just use a callable that may or may not come from `cryptograpy` package

This is just syntactic sugar because the users mostly want to use one of the function in that module, without the need of importing it (or should just use `jwk_from_pem`)

about static analyzers, yes, it's up to the users to write a good test suite for their code, or import the functions instead of using their name

Reply:
I understand.